### PR TITLE
Support SQLite for line stat SQL queries

### PR DESCRIPTION
**Pull request by SeyZ**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by SeyZ** [comment]
[Lumber users want to have support of SQLite](https://trello.com/c/tZK7VRnX/1779-lumber-users-want-to-have-support-of-sqlite)

File: services/line-stat-getter.js [hunk 1]
```diff
@@ -63,11 +63,30 @@ function LineStatGetter(model, params, opts) {
...
+        return opts.sequelize.fn('STRFTIME', '%Y-%W',
+          opts.sequelize.col(groupByDateField));
+      case 'month':
+        return opts.sequelize.fn('STRFTIME', '%Y-%m-01',
+          opts.sequelize.col(groupByDateField));
+      case 'year':
+        return opts.sequelize.fn('STRFTIME', '%Y-01-01',
+          opts.sequelize.col(groupByDateField));
+    }
+  }
```
Comment (arnaudbesnier) [inline services/line-stat-getter.js]:
```javascript
function getGroupByDateFieldFormatedForSQLite(timeRange) {
  function formatDate (format) {
    return opts.sequelize.fn('STRFTIME', format, opts.sequelize.col(groupByDateField));
  }
  switch (timeRange) {
    case 'day': return formatDate('%Y-%m-%d');
    case 'week': return formatDate('%Y-%W');
    case 'month': return formatDate('%Y-%m-01');
    case 'year': return formatDate('%Y-01-01');
  }
}
```

Comment (arnaudbesnier) [reply services/line-stat-getter.js]: 11 lines (vs 16) of code more readable and less error prone IMO

File: services/line-stat-getter.js [hunk 2]
```diff
@@ -82,6 +101,14 @@ function LineStatGetter(model, params, opts) {
     }
   }
 
+  function getGroupByDateFormat() {
+    if (timeRange === 'week' && Database.isSQLite(opts)) {
+      return [opts.sequelize.literal('"YYYY-ww"'), 'format'];
+    } else {
+      return [opts.sequelize.literal('"YYYY-MM-DD"'), 'format'];
+    }
+  }
```
Comment (arnaudbesnier) [inline services/line-stat-getter.js]: I don't really understand why this method is necessary.

Comment (arnaudbesnier) [reply services/line-stat-getter.js]: Why is it passed in the SQL query? It is the same value for each resulting rows, right?

Comment (SeyZ) [reply services/line-stat-getter.js]: I've added a comment to explain you why. Let me know if it's clear now.

**Comment by SeyZ** [comment]
@arnaudbesnier can you review it again?

File: services/line-stat-getter.js (hunk 2 above)
Comment (arnaudbesnier) [reply services/line-stat-getter.js]: Why is it passed in the SQL query If it is the same value for each resulting rows?

File: services/line-stat-getter.js [hunk 3]
```diff
@@ -172,7 +200,7 @@ function LineStatGetter(model, params, opts) {
 
   this.perform = function () {
     return model.unscoped().findAll({
-      attributes: [getGroupByDateInterval(), getAggregate()],
+      attributes: [getGroupByDateInterval(), getAggregate(), getGroupByDateFormat()],
```
Comment (arnaudbesnier) [inline services/line-stat-getter.js]: Really, there is not reason to pass an hardcoded date format in the SQL query.
Anyway it does not work on a Postgres query:
`SequelizeDatabaseError: column "YYYY-MM-DD" does not exist`

And if you run the tests, some of them fail for that reason.

Comment (SeyZ) [reply services/line-stat-getter.js]: Shit, sorry about the regression. 

If you think we don't need to hardcode de format in the SQL query, can you suggest me a solution?

File: services/line-stat-getter.js [hunk 4]
```diff
@@ -108,7 +136,7 @@ function LineStatGetter(model, params, opts) {
       records = _.sortBy(records, 'label');
       return _.map(records, function (record) {
         return {
-          label: moment(record.label).format(getFormat()),
+          label: moment(record.label, record.format).format(getFormat()),
```
Comment (arnaudbesnier) [inline services/line-stat-getter.js]: Maybe I miss something, but why don't you call the right format while formatting the label?
Once again why do you need to pass the format (which is always the same in the results)?

File: services/line-stat-getter.js [hunk 5]
```diff
@@ -108,7 +136,7 @@ function LineStatGetter(model, params, opts) {
       records = _.sortBy(records, 'label');
```
Comment (arnaudbesnier) [inline services/line-stat-getter.js]: A format seems to be missing at line 130.

File: services/line-stat-getter.js (hunk 4 above)
Comment (SeyZ) [reply services/line-stat-getter.js]: In SQLite, it's not the same for the "Group by week" use case:

- SQLite format: `YYYY-ww`
- Other databases format: `YYYY-MM-DD`

Let's sync in face-to-face this afternoon, coming to the Paris' office :)

**Pull request merged**